provisioner/powershell/provisioner.go

```diff
@@ -458,27 +469,12 @@ func (p *Provisioner) generateElevatedRunner(command string) (uploadedPath strin
 		fmt.Printf("Error creating elevated template: %s", err)
 		return "", err
 	}
-
-	tmpFile, err := ioutil.TempFile(os.TempDir(), "packer-elevated-shell.ps1")
-	writer := bufio.NewWriter(tmpFile)
-	if _, err := writer.WriteString(string(buffer.Bytes())); err != nil {
-		return "", fmt.Errorf("Error preparing elevated powershell script: %s", err)
-	}
-
-	if err := writer.Flush(); err != nil {
-		return "", fmt.Errorf("Error preparing elevated powershell script: %s", err)
-	}
-	tmpFile.Close()
-	f, err := os.Open(tmpFile.Name())
-	if err != nil {
-		return "", fmt.Errorf("Error opening temporary elevated powershell script: %s", err)
-	}
-	defer f.Close()
-
+	wrapperBytes := buffer.Bytes()
+	wrapperReader := bytes.NewReader(wrapperBytes)
```

Comment:
I think `buffer.Bytes()` already implements the reader interface, so I don't think this is needed

Reply:
Good thought, but it appears not: `cannot use buffer (type bytes.Buffer) as type io.Reader in argument to p.communicator.Upload: bytes.Buffer does not implement io.Reader (Read method has pointer receiver)`

Reply:
I would double check that

https://play.golang.org/p/5Tx2KPcucj